bin/meson-sassc

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2018 Endless Mobile, Inc.
+#
+# A small wrapper for sassc that generates a meson dependency file
+# for included partials.
+#
+# Usage: meson-sassc INPUT OUTPUT DEPFILE
+
+import argparse
+import os
+import re
+import subprocess
+import sys
+
+
+_RE_IMPORT = re.compile(r'\s*@import\s+["]([A-Za-z0-9\-_]+)["]\s*')
+
+
+def matching_lines(compiled_regex, input_fileobj):
+    '''Yield only matches of compiled_regex.'''
+    for line in input_fileobj:
+        match = compiled_regex.match(line)
+        if match != None:
+            yield match.group(1)
+
+
+def compile_sass(arg_input, arg_output, arg_depfile):
+    '''Compile sass and write depfiles.'''
+    with open(arg_input, 'r') as input_fileobj:
+        dependencies = [
+            os.path.abspath('_{}.scss'.format(line)).replace(' ', '\ ')
+            for line in matching_lines(_RE_IMPORT, input_fileobj)
+        ]
+
+    with open(arg_depfile, 'w') as depfile_fileobj:
+        depfile_fileobj.write('{arg_input}: {deps}\n'.format(arg_input=arg_input,
+                                                             deps=' '.join(dependencies)))
+
+    return subprocess.check_call(['sassc', '-a', arg_input, arg_output])
+
+
+def main(argv):
+    '''Entry point for meson-sassc.'''
+    parser = argparse.ArgumentParser(description='sassc wrapper for meson')
+    parser.add_argument('input',
+                        help='The input file',
+                        metavar='INPUT',
+                        type=str)
+    parser.add_argument('output',
+                        help='The output file',
+                        metavar='OUTPUT',
+                        type=str)
+    parser.add_argument('depfile',
+                        help='The dependency file',
+                        metavar='DEPFILE',
+                        type=str)
+    arguments = parser.parse_args(argv)
+    return compile_sass(os.path.abspath(arguments.input),
+                        os.path.abspath(arguments.output),
+                        os.path.abspath(arguments.depfile))
+
+
+if __name__ == '__main__':
+    main(sys.argv[1:])
```

Comment:
I think you have to pass the whole array to `ArgumentParser.parse_args()`, even `argv[0]`.

Reply:
In my experience passing the whole array break it. Or at least, it works when I slice off the first arg.

Reply:
I have never seen another Python program that does slices off the first argument. Are you sure?

Reply:
I have written plenty and they all work fine ;-)

However, if it works both ways, I'm happy to just take the whole argv.

Reply:
Not slicing off the first argument breaks, it, I'm afraid:

```
FAILED: data/css/wikihow.scss.css 
/sysroot/home/sam/Source/eos-knowledge-content-renderer/data/css/./meson-sassc ../data/css/wikihow.scss data/css/wikihow.scss.css data/css/sassc.deps
usage: meson-sassc [-h] INPUT OUTPUT DEPFILE
meson-sassc: error: unrecognized arguments: data/css/sassc.deps
ninja: build stopped: subcommand failed.
```

Reply:
I looked at the documentation and found the source of the confusion. You can just call `parse_args()` without passing anything in, and it will do the right thing with `sys.argv`, which is what I was thinking of. If you do pass in an array, then you are correct that you need to slice off the first element.

I'd say let's just use the no-arguments form and let it do the right thing by itself.

Reply:
Fair enough, we can also just do that. My personal preference is to pass the args explicitly (it makes `main` testable, if you ever want that), but that preference is mostly something that I do out of habit.